constructor/header.sh

```diff
@@ -609,13 +555,15 @@ if [ "$BATCH" = "0" ]; then
         printf "\\n"
     else
         case $SHELL in
-            *zsh) $PREFIX/bin/conda init zsh ;;
-            *) $PREFIX/bin/conda init ;;
+            # We call the module directly to avoid issues with spaces in shebang
+            *zsh) "$PREFIX/bin/python" -m conda init zsh ;;
+            *) "$PREFIX/bin/python" -m conda init ;;
         esac
         if [ -f "$PREFIX/bin/mamba" ]; then
             case $SHELL in
-                *zsh) $PREFIX/bin/mamba init zsh ;;
-                *) $PREFIX/bin/mamba init ;;
+                # We call the entry point directly to avoid issues with spaces in shebang
+                *zsh) "$PREFIX/bin/python" "$PREFIX/bin/mamba" init zsh ;;
+                *) "$PREFIX/bin/python" "$PREFIX/bin/mamba" init ;;
```

Comment:
```suggestion
                # We call the module directly to avoid issues with spaces in shebang
                *zsh) "$PREFIX/bin/python" -m mamba.mamba init zsh ;;
                *) "$PREFIX/bin/python" -m mamba.mamba init ;;
```

Reply:
Can you believe I always thought `-m` _required_ a package name with a `__main__` module? 😂 TIL!

I am a bit concerned the `main` function in `mamba.mamba` is removed at a later point in the future, because there's no implicit promise being made like we would understand if they had `mamba.__main__`. With that assumption, I understand the `mamba` entry point is more likely to be kept as a "module" API.

@wolfv Do you foresee any problems with using `-m mamba.mamba` as an entry point? For context, we are trying to find a way that doesn't break if the prefix contains spaces. Conda up to 4.14 will create entry points that break with spaces. Conda 22.9 contains a fix.